cmd/celestia-appd/root.go

```diff
@@ -0,0 +1,250 @@
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authcmd "github.com/cosmos/cosmos-sdk/x/auth/client/cli"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	genutilcli "github.com/cosmos/cosmos-sdk/x/genutil/client/cli"
+	"github.com/spf13/cast"
+	"github.com/spf13/cobra"
+	tmcmds "github.com/tendermint/tendermint/cmd/tendermint/commands"
+	tmcfg "github.com/tendermint/tendermint/config"
+	tmcli "github.com/tendermint/tendermint/libs/cli"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+// NewRootCmd creates a new root command for celestia-appd. It is called once in the
+// main function.
+func NewRootCmd() *cobra.Command {
+	encodingConfig := encoding.MakeEncodingConfig(app.ModuleBasics.RegisterInterfaces)
+
+	cfg := sdk.GetConfig()
+	cfg.SetBech32PrefixForAccount(app.Bech32PrefixAccAddr, app.Bech32PrefixAccPub)
+	cfg.SetBech32PrefixForValidator(app.Bech32PrefixValAddr, app.Bech32PrefixValPub)
+	cfg.SetBech32PrefixForConsensusNode(app.Bech32PrefixConsAddr, app.Bech32PrefixConsPub)
+	cfg.Seal()
+
+	initClientCtx := client.Context{}.
+		WithCodec(encodingConfig.Codec).
+		WithInterfaceRegistry(encodingConfig.InterfaceRegistry).
+		WithTxConfig(encodingConfig.TxConfig).
+		WithLegacyAmino(encodingConfig.Amino).
+		WithInput(os.Stdin).
+		WithAccountRetriever(types.AccountRetriever{}).
+		WithBroadcastMode(flags.BroadcastBlock).
+		WithHomeDir(app.DefaultNodeHome).
+		WithViper("CELESTIA")
+
+	rootCmd := &cobra.Command{
+		Use:   "celestia-appd",
+		Short: "Start celestia app",
+		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
+			initClientCtx, err := client.ReadPersistentCommandFlags(initClientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+			initClientCtx, err = config.ReadFromClientConfig(initClientCtx)
+			if err != nil {
+				return err
+			}
+
+			if err := client.SetCmdClientContextHandler(initClientCtx, cmd); err != nil {
+				return err
+			}
+
+			tmCfg := tmcfg.DefaultConfig()
+
+			customAppTemplate, customAppConfig := initAppConfig()
+			return server.InterceptConfigsPreRunHandler(cmd, customAppTemplate, customAppConfig, tmCfg)
+		},
+		SilenceUsage: true,
+	}
+
+	initRootCmd(rootCmd, encodingConfig)
+
+	return rootCmd
+}
+
+// initAppConfig helps to override default appConfig template and configs.
+// return "", nil if no custom configuration is required for the application.
+func initAppConfig() (string, interface{}) {
+	type CustomAppConfig struct {
+		serverconfig.Config
+	}
+
+	// Optionally allow the chain developer to overwrite the SDK's default
+	// server config.
+	srvCfg := serverconfig.DefaultConfig()
+	srvCfg.API.Enable = true
+	srvCfg.StateSync.SnapshotInterval = 1500
+	srvCfg.StateSync.SnapshotKeepRecent = 2
+	srvCfg.MinGasPrices = fmt.Sprintf("0%s", app.BondDenom)
```

Comment:
Can you document these?

Reply:
sure, but I'm unsure what document with as I didn't actually have any very good reason for changing these other than the default for snapshot interval is to not take any snapshots, which would reduce ssd space used by the node, but would very bad for state syncing as most nodes would not store any backups.

the default min gas price is actually insecure, because it allows for ddos of p2p network with free txs. The only reason it was changed, because the default uses `stake` instead of `celes`.

created an issue for further discussion https://github.com/celestiaorg/celestia-app/issues/413

I'm also happy to change these values back to the default too, left a comment for snapshot intervals
https://github.com/celestiaorg/celestia-app/pull/399/commits/b3eb11219eb23f6203675b7e7fefeaf80895ee95